proxy/src/main/java/net/md_5/bungee/BungeeBossBar.java

```diff
@@ -0,0 +1,284 @@
+package net.md_5.bungee;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.UUID;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import net.md_5.bungee.api.boss.BossBarColor;
+import net.md_5.bungee.api.boss.BossBarDivision;
+import net.md_5.bungee.api.boss.BossBarFlag;
+import net.md_5.bungee.api.chat.BaseComponent;
+import net.md_5.bungee.api.connection.ProxiedPlayer;
+import net.md_5.bungee.chat.ComponentSerializer;
+import net.md_5.bungee.protocol.DefinedPacket;
+import net.md_5.bungee.protocol.ProtocolConstants;
+import net.md_5.bungee.protocol.packet.BossBar;
+
+@ToString
+@EqualsAndHashCode
+public class BungeeBossBar implements net.md_5.bungee.api.boss.BossBar
+{
+
+    @Getter
+    private BaseComponent[] title;
+    @Getter
+    private BossBarColor color;
+    @Getter
+    private BossBarDivision division;
+    @Getter
+    private float health;
+    @Getter
+    private boolean visible;
+
+    private List<ProxiedPlayer> players;
+    private EnumSet<BossBarFlag> flags;
+
+    @ToString.Exclude
+    private UUID uuid;
+
+    public BungeeBossBar(BaseComponent[] title, BossBarColor color, BossBarDivision division, float health)
+    {
+        this.title = title;
```

Comment:
Please use the setXXX methods here for bounds checking and remove them in BungeeCord.createBossBar

Reply:
Wut?